### PR TITLE
PostStart annotation, trace logs on component lifecycle, docs

### PR DESCRIPTION
**Pull request by tristantarrant**
Sorry for the kitchen sink PR...

ISPN-8246 Add a PostStart annotation
    - Deprecate AbstractModuleLifecycle and add default methods to ModuleLifecycle
    - Invoke modulelifecycle.cacheManagerStarted outside of the GCR start lock
ISPN-8343 Add TRACE logs for component metadata loading and lifecycle invocations
ISPN-8332 Created and Running Cache stats should not count internal caches
ISPN-8352 EmbeddedCacheManager and RemoteCacheManager implement java.io.Closeable
Flesh out CacheManager docs
    - Add simple examples of initializing a CacheManager and describe shutdown
    - Make configuration a subchapter of the CacheManager chapter

https://issues.jboss.org/browse/ISPN-8246
https://issues.jboss.org/browse/ISPN-8343
https://issues.jboss.org/browse/ISPN-8332
https://issues.jboss.org/browse/ISPN-8352

### CODE REVIEW

**Comment by tristantarrant** [comment]
Please check the changes to the user-guide here: http://www.dataforte.net/listing/infinispan/generated-html/user_guide/user_guide.html#the_embedded_cachemanager

File: core/src/main/java/org/infinispan/factories/annotations/PostStart.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Method level annotation that indicates a (no-param) method to be called on a component after the
+ * {@link org.infinispan.factories.ComponentRegistry} has been fully initialized
+ * <p/>
+ *
+ * @author Tristan Tarrant
+ * @since 9.1
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/factories/annotations/PostStart.java]: 9.2

**Comment by tristantarrant** [comment]
@pruivo addressed all comments

**Comment by pruivo** [comment]
@tristantarrant there are some checkstyle errors in CI

**Comment by tristantarrant** [comment]
Yeah, waiting to merge the tx revert so I can rebase and have CI run again

**Comment by tristantarrant** [comment]
Rebased

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant

**Pull request merged**